Support newer Kotlin versions for compiling against IntelliJ IDEA 2025.3.
<+>UTF-8
===================================================================
diff --git a/kotlin/internal/toolchains.bzl b/kotlin/internal/toolchains.bzl
index b69a16a1..a6df8f05 100644
--- a/kotlin/internal/toolchains.bzl
+++ b/kotlin/internal/toolchains.bzl
@@ -144,6 +144,8 @@ _kt_toolchain = rule(
                 "1.9",
                 "2.0",
                 "2.1",
+                "2.2",
+                "2.3",
             ],
         ),
         "api_version": attr.string(
@@ -161,6 +163,8 @@ _kt_toolchain = rule(
                 "1.9",
                 "2.0",
                 "2.1",
+                "2.2",
+                "2.3",
             ],
         ),
         "debug": attr.string_list(
diff --git a/src/main/starlark/core/options/opts.kotlinc.bzl b/src/main/starlark/core/options/opts.kotlinc.bzl
index 14069701..8aa097a2 100644
--- a/src/main/starlark/core/options/opts.kotlinc.bzl
+++ b/src/main/starlark/core/options/opts.kotlinc.bzl
@@ -430,7 +430,7 @@ default: 'first-only-warn' in language version 2.2+, 'first-only' in version 2.1
         args = dict(
             default = "",
             doc = "The target version of the generated JVM bytecode",
-            values = ["1.6", "1.8", "9", "10", "11", "12", "13", "15", "16", "17"],
+            values = ["1.6", "1.8", "9", "10", "11", "12", "13", "15", "16", "17", "21"],
         ),
         type = attr.string,
         value_to_flag = None,
